### PR TITLE
Bridge gaps between content serializer features

### LINKED ISSUE #3951: Add the retrieve logic to MultipleArtifactSerializer

### PR DESCRIPTION
**Pull request by mdellweg**
Refactor the content serializers to reuse more code and share more features.
* Allow to implement 'retrieve' on all content serializers for multiple upload.
* Allow adding all content to a repository on creation.
* Validate that the content type matches the provided repository.

fixes #3951

### CODE REVIEW

File: pulpcore/plugin/viewsets/content.py [hunk 1]
```diff
@@ -43,20 +43,19 @@ def create(self, request):
...
         temp_file = PulpTemporaryFile.init_and_validate(file_content)
         temp_file.save()
 
-        resources = []
-        repository = serializer.validated_data.get("repository")
-        if repository:
-            resources.append(repository)
+        exclusive_resources = [
+            item for item in (serializer.validated_data.get(key) for key in ("repository")) if item
+        ]
```
Comment (lubosmj) [inline pulpcore/plugin/viewsets/content.py:53]:
```suggestion
        exclusive_resources = [
            item for item in (serializer.validated_data.get(key) for key in ("repository",)) if item
        ]
```
You are iterating over the list of characters. Is this complex list comprehension necessary?

Comment (mdellweg) [reply pulpcore/plugin/viewsets/content.py:53]: Thanks! Not it's probably a bit much.

Comment (mdellweg) [reply pulpcore/plugin/viewsets/content.py]: I wanted to bring the two viewsets closer together in the way they are implemented.

**Review by lubosmj — APPROVED** [review]
Neat!

**Review by dkliban — APPROVED** [review]

**Pull request merged**